Compilation failure with GCC 14.1

- sst-core current devel (c91f81eb08acdd3fa4f249acd55ac202870778b7)
- configure without MPI and without ncurses
- OS: Arch Linux with gcc 14.1.1 20240522, but likely GCC 14.1 on any distribution will fail

```console
/home/eric/development/sst/forks/sst-core/src/sst/core/sstinfo.cc: In member function ‘void SST::SSTLibraryInfo::outputText(std::stringstream&)’:
/home/eric/development/sst/forks/sst-core/src/sst/core/sstinfo.cc:861:30: error: no matching function for call to ‘find(std::vector<std::__cxx11::basic_string<char> >::iterator, std::vector<std::__cxx11::basic_string<char> >::iterator, std::string&)’
  861 |                     std::find(m_componentFilters.begin(), m_componentFilters.end(), component.componentName) !=
      |                     ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/c++/14.1.1/bits/locale_facets.h:48,
                 from /usr/include/c++/14.1.1/bits/basic_ios.h:37,
                 from /usr/include/c++/14.1.1/ios:46,
                 from /usr/include/c++/14.1.1/ostream:40,
                 from /usr/include/c++/14.1.1/iostream:41,
                 from /home/eric/development/sst/forks/sst-core/src/sst/core/configBase.h:19,
                 from /home/eric/development/sst/forks/sst-core/src/sst/core/configShared.h:15,
                 from /home/eric/development/sst/forks/sst-core/src/sst/core/sstinfo.h:15,
                 from /home/eric/development/sst/forks/sst-core/src/sst/core/sstinfo.cc:14:
/usr/include/c++/14.1.1/bits/streambuf_iterator.h:435:5: note: candidate: ‘template<class _CharT2> typename __gnu_cxx::__enable_if<std::__is_char<_CharT2>::__value, std::istreambuf_iterator<_CharT, std::char_traits<_CharT> > >::__type std::find(istreambuf_iterator<_CharT, char_traits<_CharT> >, istreambuf_iterator<_CharT, char_traits<_CharT> >, const _CharT2&)’
  435 |     find(istreambuf_iterator<_CharT> __first,
      |     ^~~~
/usr/include/c++/14.1.1/bits/streambuf_iterator.h:435:5: note:   template argument deduction/substitution failed:
/home/eric/development/sst/forks/sst-core/src/sst/core/sstinfo.cc:861:30: note:   ‘__gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char>*, std::vector<std::__cxx11::basic_string<char> > >’ is not derived from ‘std::istreambuf_iterator<_CharT, std::char_traits<_CharT> >’
  861 |                     std::find(m_componentFilters.begin(), m_componentFilters.end(), component.componentName) !=
      |                     ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/eric/development/sst/forks/sst-core/src/sst/core/sstinfo.cc: In member function ‘void SST::SSTLibraryInfo::filterSearch(std::stringstream&, std::string, std::string)’:
/home/eric/development/sst/forks/sst-core/src/sst/core/sstinfo.cc:903:35: error: no matching function for call to ‘find(std::vector<std::__cxx11::basic_string<char> >::iterator, std::vector<std::__cxx11::basic_string<char> >::iterator, std::string&)’
  903 |                     if ( std::find(stringList.begin(), stringList.end(), infoTag) != stringList.end() ) {
      |                          ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/usr/include/c++/14.1.1/bits/streambuf_iterator.h:435:5: note: candidate: ‘template<class _CharT2> typename __gnu_cxx::__enable_if<std::__is_char<_CharT2>::__value, std::istreambuf_iterator<_CharT, std::char_traits<_CharT> > >::__type std::find(istreambuf_iterator<_CharT, char_traits<_CharT> >, istreambuf_iterator<_CharT, char_traits<_CharT> >, const _CharT2&)’
  435 |     find(istreambuf_iterator<_CharT> __first,
      |     ^~~~
/usr/include/c++/14.1.1/bits/streambuf_iterator.h:435:5: note:   template argument deduction/substitution failed:
/home/eric/development/sst/forks/sst-core/src/sst/core/sstinfo.cc:903:35: note:   ‘__gnu_cxx::__normal_iterator<std::__cxx11::basic_string<char>*, std::vector<std::__cxx11::basic_string<char> > >’ is not derived from ‘std::istreambuf_iterator<_CharT, std::char_traits<_CharT> >’
  903 |                     if ( std::find(stringList.begin(), stringList.end(), infoTag) != stringList.end() ) {
      |                          ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/eric/development/sst/forks/sst-core/src/sst/core/sstinfo.cc: In member function ‘void SST::SSTLibraryInfo::clearHighlights()’:
/home/eric/development/sst/forks/sst-core/src/sst/core/sstinfo.cc:957:53: error: cannot convert ‘std::__cxx11::basic_string<char>::iterator’ to ‘const char*’
  957 |                 infoStr->erase(remove(infoStr->begin(), infoStr->end(), '`'), infoStr->end());
      |                                       ~~~~~~~~~~~~~~^~
      |                                                     |
      |                                                     std::__cxx11::basic_string<char>::iterator
In file included from /usr/include/c++/14.1.1/cstdio:42,
                 from /usr/include/c++/14.1.1/ext/string_conversions.h:45,
                 from /usr/include/c++/14.1.1/bits/basic_string.h:4154,
                 from /usr/include/c++/14.1.1/string:54,
                 from /usr/include/c++/14.1.1/bits/locale_classes.h:40,
                 from /usr/include/c++/14.1.1/bits/ios_base.h:41,
                 from /usr/include/c++/14.1.1/ios:44:
/usr/include/stdio.h:158:32: note:   initializing argument 1 of ‘int remove(const char*)’
  158 | extern int remove (const char *__filename) __THROW;
      |                    ~~~~~~~~~~~~^~~~~~~~~~
```

--- a/src/sst/core/sstinfo.cc
+++ b/src/sst/core/sstinfo.cc
@@ -23,6 +23,7 @@
 #include "sst/core/subcomponent.h"
 #include "sst/core/warnmacros.h"
 
+#include <algorithm>
 #include <cerrno>
 #include <cstdio>
 #include <cstdlib>
@@ -180,7 +181,7 @@ main(int argc, char* argv[])
 std::string
 convertToLower(std::string input)
 {
-    transform(input.begin(), input.end(), input.begin(), ::tolower);
+    std::transform(input.begin(), input.end(), input.begin(), ::tolower);
     return input;
 }
 

